Modules/_pickle.c

```diff
@@ -1382,11 +1382,13 @@ _Unpickler_ResizeMemoList(UnpicklerObject *self, Py_ssize_t new_size)
 
     assert(new_size > self->memo_size);
 
-    PyMem_RESIZE(self->memo, PyObject *, new_size);
-    if (self->memo == NULL) {
+    PyObject **memo_new = self->memo;
+    PyMem_Resize(memo_new, PyObject *, new_size);
```

Comment:
AFAIK currently there is no difference between `PyMem_RESIZE` and `PyMem_Resize`. But since this PR will be backported to old versions, I suggest to use `PyMem_RESIZE`.